diff --git a/drivers/mmc/host/sdhci-of-dwcmshc.c b/drivers/mmc/host/sdhci-of-dwcmshc.c
index 3bc2cdc996a1b..036631f5615b3 100644
--- a/drivers/mmc/host/sdhci-of-dwcmshc.c
+++ b/drivers/mmc/host/sdhci-of-dwcmshc.c
@@ -528,6 +528,12 @@ static int hailo15_dwcmshc_execute_tuning(struct mmc_host *mmc, u32 opcode)
 	err = sdhci_execute_tuning(mmc, opcode);
 	if (!err && !host->tuning_err)
 		pr_info("%s Tuning Success!\n", mmc_hostname(host->mmc));
+	else {
+		// MSW-6198 - In case of an Error - try one more time (full iteration loop)
+		err = sdhci_execute_tuning(mmc, opcode);
+		if (!err && !host->tuning_err)
+			pr_info("%s Tuning Success!\n", mmc_hostname(host->mmc));
+	}
 
 	return err;
 }
@@ -764,12 +770,16 @@ static void dwcmshc_hailo15_phy_config(struct sdhci_host *host)
 	reg16 |= DWCMSHC_XFER_MODE_R__MULTI_BLK_SEL;
 	sdhci_writew(host, reg16, DWCMSHC_XFER_MODE_R);
 
-	/* Sampling window threshold value is 31 out of 128 taps */
+	/* Sampling window threshold value */
 	reg32 = sdhci_readl(host, DWCMSHC_AT_CTRL_R);
 	reg32 &= ~DWCMSHC_AT_CTRL_R__SWIN_TH_EN;
 	reg32 |= DWCMSHC_AT_CTRL_R__SWIN_TH_EN;
 	reg32 &= ~DWCMSHC_AT_CTRL_R__SWIN_TH_VAL;
-	reg32 |= FIELD_PREP(DWCMSHC_AT_CTRL_R__SWIN_TH_VAL,0x1f);
+	if (sdio_phy_config->card_is_emmc) {
+		reg32 |= FIELD_PREP(DWCMSHC_AT_CTRL_R__SWIN_TH_VAL,0x3c);
+	} else {
+		reg32 |= FIELD_PREP(DWCMSHC_AT_CTRL_R__SWIN_TH_VAL,0x1f);
+	}
 	sdhci_writel(host, reg32, DWCMSHC_AT_CTRL_R);
 
 	pr_debug("%s phy configuration for %s mode done\n", mmc_hostname(host->mmc), sdio_phy_config->card_is_emmc ? "EMMC ": "SD");
diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index cf4524a97e37e..b0d7bbbdb18ac 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -2809,7 +2809,7 @@ static int __sdhci_execute_tuning(struct sdhci_host *host, u32 opcode)
 
 	}
 
-	pr_debug("%s: Tuning failed, falling back to fixed sampling clock\n",
+	pr_err("%s: Tuning failed, falling back to fixed sampling clock\n",
 		mmc_hostname(host->mmc));
 	sdhci_reset_tuning(host);
 	return -EAGAIN;
